sources/packages/forms/src/form-definitions/educationprogram.json

```diff
@@ -1691,12 +1693,12 @@
                     "tags": [],
                     "properties": {},
                     "conditional": {
-                        "show": true,
-                        "when": "programDeliveryTypes",
-                        "eq": "deliveredOnline",
+                        "show": "",
+                        "when": null,
+                        "eq": "",
                         "json": ""
                     },
-                    "customConditional": "",
+                    "customConditional": "const isBCInstitution = data.isBCPublic || data.isBCPrivate;\nshow = (isBCInstitution === false && data.programDeliveryTypes.deliveredOnline);",
```

Comment:
I tried this code, but it would cause a backend error when "Online" is selected for BC institutions. I then used `isBCInstitution === false` instead.

Reply:
@lewischen-aot regarding the backend error, when there is something needed from the server side, for instance, the institution type, we must ensure that the server will populate the payload to avoid user manipulation. We should not rely on "the user" providing the institution type.
We did the same here:
https://github.com/bcgov/SIMS/blob/main/sources/packages/backend/apps/api/src/route-controllers/designation-agreement/designation-agreement.institutions.controller.ts#L96-L102

Reply:
Thanks @andrewsignori-aot @lewischen-aot and I were discussing about it. We think that's the best way to solve that.

Reply:
Thank you so much @andrewsignori-aot . This is the solution I was looking for!

Reply:
I learned something new today 😄 When sending payloads to the backend, not all fields in the frontend payload need match frontend DTOs even though the fields are declared in the frontend DTO. Frontend and backend DTOs need to match, and payloads can be validated in backend DTOs. As long as the payload fields do not have `@Allow` decorators in the backend DTO, these fields will not be validated in the Nestjs for databases. Sorry it's not very accurate, but that's roughly the idea 😊

Reply:
I am good with it. 👍

Reply:
@lewischen-aot I believe that it is mostly accurate, just to ensure that we are on the same page.
- API is supposed to reject any payload with unknown properties. It is configurable on Nestjs. Extra properties can be ignored or generate an error and we are using this last approach due to some Ministry security scams that required it.
- Web and API DTOs must match and for dynamic forms, the Web project has a special method to "clean" the form.io submission from anything not expected in the API.
- The interfaces that map form.io are an attempt to give some type structure, but at runtime form.io can receive and send more data than defined. It is up to us to try to enforce the typing.

Reply:
Thanks for the comment @andrewsignori-aot. I totally agree. We are on the same page 👍